tf2_ros/include/tf2_ros/async_buffer_interface.h

```diff
@@ -47,7 +47,7 @@ namespace tf2_ros
 
 class TransformStampedFuture : public std::shared_future<geometry_msgs::msg::TransformStamped>
 {
-  typedef std::shared_future<geometry_msgs::msg::TransformStamped> _Base_type;
+  using _Base_type = std::shared_future<geometry_msgs::msg::TransformStamped>;
```

Comment:
Actually, the `_Base_type` identifier is not C++ standard compliant. From [Identifiers (cppreference)](https://en.cppreference.com/w/cpp/language/identifiers):
```the identifiers that begin with an underscore followed by an uppercase letter are reserved```

How about:

```cpp
using Base_Type = std::shared_future<geometry_msgs::msg::TransformStamped>;
```

The `Base_Type` type alias is scoped within `tf2_ros::TransformStampedFuture` anyway, so there wouldn't be any identifier conflicts.

Reply:
Thank you. Updated in https://github.com/ros2/geometry2/pull/490/commits/d30e1878376b150177ab73152a4dfcd223284a46

Reply:
I need to say, it's from `/usr/include/c++/9/future`
```cpp
  /// Primary template for shared_future.
  template<typename _Res>
    class shared_future : public __basic_future<_Res>
    {
      typedef __basic_future<_Res> _Base_type;

    public:
```